presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -473,18 +470,29 @@ private void invokeNoMoreSplitsIfNecessary()
             }
 
             // Create a registry of delete_delta directories for the partition
-            AcidInfo.Builder acidInfoBuilder = AcidInfo.builder(path);
             for (AcidUtils.ParsedDelta delta : directory.getCurrentDirectories()) {
                 if (delta.isDeleteDelta()) {
                     acidInfoBuilder.addDeleteDelta(delta.getPath(), delta.getMinWriteId(), delta.getMaxWriteId(), delta.getStatementId());
                 }
             }
 
-            acidInfo = acidInfoBuilder.build();
+            // initialize original files status list if present
+            fileStatusOriginalFiles = directory.getOriginalFiles();
+
+            for (HadoopShims.HdfsFileStatusWithId hdfsFileStatusWithId : fileStatusOriginalFiles) {
+                Path originalFilePath = new Path(hdfsFileStatusWithId.getFileStatus().getPath().toUri().getPath());
+                long originalFileLength = hdfsFileStatusWithId.getFileStatus().getLen();
+
+                if (originalFileLength == 0) {
+                    continue;
+                }
+
+                int bucketId = getBucketNumber(originalFilePath.getName()).getAsInt();
```

Comment:
```suggestion
                int bucketId = getBucketNumber(originalFilePath.getName())
                        .orElseThrow(() -> new IllegalStateException("Cannot get bucket number from path: " + originalFilePath));
```

Reply:
However, what if tables was not a bucketed table and was converted into a transactional one?

Reply:
We are fetching bucketID which we get from the original file name, this will always be needed irrespective of the table is bucketed or not. We need this bucket id to identify the correct delete delta with the same bucket.